plugins/jobtype/src/azkaban/jobtype/tuning/TuningErrorHandler.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2012 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package azkaban.jobtype.tuning;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.apache.log4j.Logger;
+
+/**
+ * This class is responsible for finding whether failure is because of tuning parameters.
+ * This try to search predefined patterns in the log.
+ */
+public class TuningErrorHandler {
+  private Logger log = Logger.getRootLogger();
+
+  List<Pattern> errorPatterns = new ArrayList<Pattern>(3);
```

Comment:
Why bother with an initial capacity when you are only adding 4 elements? Also, the initial capacity isn't even correct based on how many elements you will actually add.

This could easily be a `static final`, and add the patterns in a static initializer block. There is no need for each instance to have its own copy of this.

Reply:
Done.